### PR TITLE
Fluid Regulator GUI fixes

### PR DESCRIPTION
**Pull request by miozune**
- Fix textbox does not accept negative value
- Fix clicking import/export button does not update textbox value
- Fix inputting invalid values in a row does not update textbox
- Fix messing around with tick rate can bypass pump limit

### CODE REVIEW

File: src/main/java/gregtech/common/covers/GT_Cover_FluidRegulator.java [hunk 1]
```diff
@@ -456,8 +471,6 @@ public void applyTextBox(GT_GuiIntegerTextBox box) {
 				if (i > maxFlow) {
 					i = maxFlow;
 					warn = true;
-				} else if (i < 0) {
```
Comment (GlodBlock) [inline src/main/java/gregtech/common/covers/GT_Cover_FluidRegulator.java:459]: you need to limit the minimum negative value, you can set -9999999L/s on lv fluid regulator now

File: src/main/java/gregtech/common/covers/GT_Cover_FluidRegulator.java [hunk 2]
```diff
@@ -368,11 +368,21 @@ public GUI(byte aSide, int aCoverID, FluidRegulatorData aCoverVariable, ICoverab
 			new GT_GuiIconButton(this, 3, startX + spaceX * 1, startY + spaceY * 1, GT_GuiIcon.REDSTONE_ON).setTooltipText(GT_Utility.trans("225", "Active with Redstone Signal"));
 			new GT_GuiIconButton(this, 4, startX + spaceX * 2, startY + spaceY * 1, GT_GuiIcon.REDSTONE_OFF).setTooltipText(GT_Utility.trans("226", "Inactive with Redstone Signal"));
 
-			tBox = new GT_GuiIntegerTextBox(this, 2, startX + spaceX * 0, startY + spaceY * 2 + 2, spaceX * 4 - 3, 12);
+			tBox = new GT_GuiIntegerTextBox(this, 2, startX + spaceX * 0, startY + spaceY * 2 + 2, spaceX * 4 - 3, 12) {
+                @Override
+                public boolean validChar(char c, int key) {
+                    return super.validChar(c, key) || c == '-';
```
Comment (GlodBlock) [inline src/main/java/gregtech/common/covers/GT_Cover_FluidRegulator.java:374]: people can write `-` anywhere like `-12-1` in it, thought it doesn't affect game

**Review by GlodBlock — CHANGES_REQUESTED** [review]

File: src/main/java/gregtech/common/covers/GT_Cover_FluidRegulator.java (hunk 1 above)
Comment (miozune) [reply src/main/java/gregtech/common/covers/GT_Cover_FluidRegulator.java:459]: Ah, carelessness

File: src/main/java/gregtech/common/covers/GT_Cover_FluidRegulator.java (hunk 2 above)
Comment (miozune) [reply src/main/java/gregtech/common/covers/GT_Cover_FluidRegulator.java:374]: Yes invalid inputs are just reverted, so it won't hurt

**Comment by miozune** [comment]
Pushed some more commits to address review and fix more issues.

**Review by GlodBlock — APPROVED** [review]

**Pull request merged**